src/init-ceph.in

```diff
@@ -31,11 +31,22 @@ if [ `dirname $0` = "." ] && [ $PWD != "/etc/init.d" ]; then
     SYSTEMD_RUN=""
     ASSUME_DEV=1
 else
-    BINDIR=@bindir@
-    SBINDIR=@prefix@/sbin
-    LIBEXECDIR=@libexecdir@/ceph
-    ETCDIR=@sysconfdir@/ceph
-    ASSUME_DEV=0
+    if [ `dirname $0` = "bin" ] && [ -e CMakeCache.txt ]; then
```

Comment:
can we just check `[ -e CMakeCache.txt ]`?
i was using `./bin/init-ceph restart osd.0`, and it failed to detect that i was under a build directory of cmake.